frontend/src/metabase/lib/expressions/resolver.js

```diff
@@ -35,7 +37,12 @@ function findMBQL(op) {
   return clause;
 }
 
-export function resolve(expression, type, fn) {
+const isCompatible = (a, b) =>
+  a === b ||
```

Comment:
nit(?): These are three semantically different cases so it feels iffy putting them in a single expression. If statements without braces are also "not great" but I think in this case might make it clearer that there are three different cases to look for.

Reply:
Good point, thanks! I've changed it so the code becomes a bit more verbose, but I agree that clarity is important here.